fabric-particles-v1/src/main/java/net/fabricmc/fabric/impl/particles/ParticleFactoryRegistryImpl.java

```diff
@@ -72,16 +73,14 @@ public void injectValues(Int2ObjectMap<ParticleFactory<?>> factories, Map<Identi
                 return constr;
             }
 
-            for (Class<?> cls : ParticleManager.class.getDeclaredClasses()) {
-                if (!cls.isInterface() && SpriteProvider.class.isAssignableFrom(cls)) {
-                    constr = cls.getDeclaredConstructor(ParticleManager.class);
-                    constr.setAccessible(true);
+            String className = FabricLoader.getInstance().getMappingResolver().mapClassName("intermediary", "net.minecraft.class_702$class_4090");
```

Comment:
You should add a comment so you know exactly what this is without needing to look it up.

Reply:
`throw new IllegalStateException("net.minecraft.client.particle.ParticleManager.SimpleSpriteProvider.<init>() is gone!", e);`

Reply:
`Uses reflection to obtain new instances of the private inner class `ParticleManager.SimpleSpriteProvider`.`